consensus/src/liveness/proposal_generator.rs

```diff
@@ -345,6 +345,10 @@ impl ProposalGenerator {
                 .await
                 .context("Fail to retrieve payload")?;
 
+            if !payload.is_direct() && payload.len() > max_txns_from_block_to_execute as usize {
```

Comment:
in addition to this, I would prefer if max_txns_from_block_to_execute was Option itself, so if we don't want to limit it, we don't need to rely on value being large enough, but can just set it to None